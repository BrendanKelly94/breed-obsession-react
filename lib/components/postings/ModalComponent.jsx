import React from 'react';

const ModalComponent = ({ selectedPost, closeModal}) => {

  return (
    <div>
      <div className = "modal-header">
        <h4 className = "modal-title">{selectedPost.name}</h4>
      </div>
      <div className = "modal-body">
        <div className = "row container">

        </div>
        <div className = "row container modal-details">
          <p> <b>Status:</b> {selectedPost.status} </p>
          <p> <b>Age:</b> {selectedPost.age} </p>
          <p> <b>Sex:</b> {selectedPost.sex} </p>
          <p> <b>Breed:</b> {selectedPost.breed} </p>
          <p> <b>Location:</b> {selectedPost.city} , {selectedPost.state} </p>
          <p> <b>Email:</b> {selectedPost.email} </p>
          <p> <b>Phone:</b> {selectedPost.phone} </p>
        </div>

        <div className = "row container">
          {selectedPost.description}
        </div>
      </div>
      <div className = "modal-footer">
        <button type = "button" className = "btn btn-outline-dark" onClick = {() => closeModal()} >Close</button>
      </div>
    </div>
  );
};

export default ModalComponent;
