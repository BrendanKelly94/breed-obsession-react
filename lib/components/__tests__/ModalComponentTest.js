import React from 'react';
import ModalComponent from '../postings/ModalComponent.jsx';

import renderer from 'react-test-renderer';

describe('PostComponent', () => {

  const testProps = {
    selectedPost: {},
    closeModal: jest.fn(() => ({}))
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <ModalComponent
        {...testProps}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();

  });
});
