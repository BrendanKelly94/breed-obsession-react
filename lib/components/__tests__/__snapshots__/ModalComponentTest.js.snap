// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostComponent renders correctly 1`] = `
<div>
  <div
    className="modal-header"
  >
    <h4
      className="modal-title"
    />
  </div>
  <div
    className="modal-body"
  >
    <div
      className="row container"
    />
    <div
      className="row container modal-details"
    >
      <p>
         
        <b>
          Status:
        </b>
         
         
      </p>
      <p>
         
        <b>
          Age:
        </b>
         
         
      </p>
      <p>
         
        <b>
          Sex:
        </b>
         
         
      </p>
      <p>
         
        <b>
          Breed:
        </b>
         
         
      </p>
      <p>
         
        <b>
          Location:
        </b>
         
         , 
         
      </p>
      <p>
         
        <b>
          Email:
        </b>
         
         
      </p>
      <p>
         
        <b>
          Phone:
        </b>
         
         
      </p>
    </div>
    <div
      className="row container"
    />
  </div>
  <div
    className="modal-footer"
  >
    <button
      className="btn btn-outline-dark"
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;
